Content.Server/Doors/Systems/FirelockSystem.cs

```diff
@@ -136,8 +142,17 @@ private void OnBeforeDoorOpened(EntityUid uid, FirelockComponent component, Befo
             // Give the Door remote the ability to force a firelock open even if it is holding back dangerous gas
             var overrideAccess = (args.User != null) && _accessReaderSystem.IsAllowed(args.User.Value, uid);
 
-            if (!this.IsPowered(uid, EntityManager) || (!overrideAccess && IsHoldingPressureOrFire(uid, component)))
+            var isPowered = this.IsPowered(uid, EntityManager);
+            if (!isPowered || (!overrideAccess && IsHoldingPressureOrFire(uid, component)))
+            {
+                if (args.User != null && !isPowered)
+                {
+                    _popupSystem.PopupEntity(Loc.GetString("firelock-component-is-unpowered-message"),
+                    uid, args.User.Value, PopupType.Medium);
+                }
+
                 args.Cancel();
+            }
```

Comment:
Can you simplify this and make it a guard pattern (e.g. Check power by itself first) so that it's easier to read?